pcdsdevices/targets.py

```diff
@@ -260,3 +267,786 @@ def move(self, nxspaces, nyspaces, wait=False):
+        from the middle of the four targets that are encasing the grid.
+
+        The user will be asked to define the coordinates using the `tweak`
+        method.
+
+        Examples
+        --------
+        # Press q when ready to save the coordinates
+        >>> xy.set_presets()
+        Setting coordinates for (0, 0) top left corner:
+
+        0.0000, : 0.0000, scale: 0.1
+
+        Setting coordinates for (0, M) top right corner:
+
+        10.0000, : 0.0000, scale: 0.1
+
+        Setting coordinates for (N, M) bottom right corner:
+
+        10.0000, : -10.0000, scale: 0.1
+
+        Setting coordinates for (N, 0) bottom left corner:
+
+        -0.0000, : -10.0000, scale: 0.1
+        """
+        # check to see the the presets are setup
+        if not hasattr(self.x.presets, 'add_hutch'):
+            raise AttributeError('No folder setup for motor presets. '
+                                 'Please add a location to save the positions '
+                                 'to, using setup_preset_paths from '
+                                 'pcdsdevices.interface to save the position.')
+
+        print('\nSetting coordinates for (0, 0) top left corner: \n')
+        self.tweak()
+        pos = [self.x.position, self.y.position]
+        print('\nSetting coordinates for (0, M) top right corner: \n')
+        self.tweak()
+        pos.extend([self.x.position, self.y.position])
+        print('\nSetting coordinates for (N, M) bottom right corner: \n')
+        self.tweak()
+        pos.extend([self.x.position, self.y.position])
+        print('\nSetting coordinates for (N, 0) bottom left corner: \n')
+        self.tweak()
+        pos.extend([self.x.position, self.y.position])
+        # create presets
+        # corner (0, 0)
+        self.x.presets.add_hutch(value=pos[0], name="x_top_left")
+        self.y.presets.add_hutch(value=pos[1], name="y_top_left")
+        # corner (0, M)
+        self.x.presets.add_hutch(value=pos[2], name="x_top_right")
+        self.y.presets.add_hutch(value=pos[3], name="y_top_right")
+        # corner (M, N)
+        self.x.presets.add_hutch(value=pos[4], name="x_bottom_right")
+        self.y.presets.add_hutch(value=pos[5], name="y_bottom_right")
+        # corner (N, 0)
+        self.x.presets.add_hutch(value=pos[6], name="x_bottom_left")
+        self.y.presets.add_hutch(value=pos[7], name="y_bottom_left")
+
+    def get_presets(self):
+        """
+        Get the saved presets if any.
+
+        Examples
+        --------
+        >>> xy.get_presets()
+        ((0, 0),
+        (9.99999999999998, 0),
+        (9.99999999999998, -9.99999999999998),
+        (-6.38378239159465e-16, -9.99999999999998))
+
+        Returns
+        -------
+        coord : tuple
+            Four coordinate positions.
+            (top_left, top_right, bottom_right, bottom_left)
+        """
+        try:
+            # corner (0, 0)
+            top_left = (round(self.x.presets.positions.x_top_left.pos, 5),
```

Comment:
I probably should not round this guys here... but I am not getting exactly the same points back.

```
In [28]: stage.set_presets()

Setting coordinates for (0, 0) top left corner:

: 0.0000, : 0.0000, scale: 0.1

Setting coordinates for (0, M) top right corner:

: 0.0000, : 2.0000, scale: 0.1

Setting coordinates for (N, M) bottom right corner:

: 2.0000, : 2.0000, scale: 0.1

Setting coordinates for (N, 0) bottom left corner:

: 2.0000, : 1.0000, scale: 0.1
```

```
In [29]: stage.get_presets()
Out[29]:
((0, 0),
 (0, 2.0000000000000004),
 (2.0000000000000004, 2.0000000000000004),
 (2.0000000000000004, 0.9999999999999997))
```

Reply:
Rounding for display purposes is just fine (encouraged, even), but rounding prior to doing calculations should be avoided.

Reply:
ok thanks, it'll take those off then...